GitCommands/Git/GitModule.cs

```diff
@@ -1508,11 +1508,34 @@ public string FormatPatch(string from, string to, string output)
             return result;
         }
 
-        public string Tag(string tagName, string revision, bool annotation, bool force)
-        {
-            if (annotation)
-                return RunGitCmd(string.Format("tag \"{0}\" -a {1} -F \"{2}\\TAGMESSAGE\" -- \"{3}\"", tagName.Trim(), (force ? "-f" : ""), GetGitDirectory(), revision));
-            return RunGitCmd(string.Format("tag {0} \"{1}\" \"{2}\"", (force ? "-f" : ""), tagName.Trim(), revision));
+        public string Tag(string tagName, string revision, bool force, int tagIdx = 0, string keyId = "")
+        {
+            string strCommand = "";
+
+            switch (tagIdx)
```

Comment:
Instead of `tagIdx` I'd call this something like `operation` or `operationType`.
I wonder whether it is worth introducing an enum and use it instead of using magic numbers.